File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02660/p02660_num8_parsed.py
import math

def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def count_operations(n):
    primes = []
    operations = 0
    while n > 1:
        found_prime = False
        for i in range(2, int(math.sqrt(n)) + 1):
            if is_prime(i) and n % i == 0:
                p = i
                e = 1
                while n % (p ** (e + 1)) == 0:
                    e += 1
                z = p ** e
                if z not in primes:
                    primes.append(z)
                    found_prime = True
                    break
        if found_prime:
            n //= z
            operations += 1
        else:
            break
    return operations

# Read input N
N = int(input())

# Calculate the maximum number of operations
max_operations = count_operations(N)

# Print the maximum number of operations
print(max_operations)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02660/p02660_num8_parsed.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

// Function to check if a number is prime
bool isPrime(int n) {
    if (n <= 1)
        return false;
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0)
            return false;
    }
    return true;
}

// Function to get the maximum number of times the operation can be applied
int getMaxOperations(int N) {
    vector<int> operations;
    int count = 0;
    for (int i = 2; i <= sqrt(N); i++) {
        if (N % i == 0 && isPrime(i)) {
            operations.push_back(i);
            N /= i;
            count++;
        }
    }
    if (N > 1) {
        operations.push_back(N);
        count++;
    }
    return count;
}

int main() {
    int N;
    cin >> N;
    int maxOperations = getMaxOperations(N);
    cout << maxOperations << endl;
    return 0;
}